### PR TITLE
Documentation update

### LINKED ISSUE #72: Examples are not numbered

### LINKED ISSUE #73: Top-level clauses and annexes are incorrectly references
_5. Hello World_ should be referenced as _Clause 5_ and not _5_.

### PR DESCRIPTION
**Pull request by SteveLLamb**
First pass documentation update for sections and elements used in AGs

Closes #72
Closes #73

### CODE REVIEW

File: doc/main.html [hunk 1]
```diff
@@ -200,13 +255,17 @@ <h4>introduction section</h4>
 
         <p>The <code>id</code> attribute shall be present and equal to <code>sec-introduction</code>.</p>
 
+        <p>If no Introduction is required, the section shall not be present.</p>
```
Comment (palemieux) [inline doc/main.html:258]:
```suggestion
        <p>The absence of a section indicates that the author did not provide introductory text.</p>
```

I am not sure we should prohibit an empty section.

File: doc/main.html [hunk 2]
```diff
@@ -148,50 +150,103 @@ <h4>Overall structure</h4>
...
       </section>
 
       <section id="sec-introduction-section">
-        <h4>introduction section</h4>
+        <h4>Introduction section</h4>
+
+        <p>The <dfn>Introduction</dfn> contains the introduction of the document.</p>
 
-        <p>The <dfn>introduction section</dfn> contains the introduction of the document.</p>
+          <p>This section is optional. If using this section, it shall be present as: </p>
```
Comment (palemieux) [inline doc/main.html:249]:
```suggestion
          <p>Below is an example:</p>
```

I do not think we can say "shall be" since we might have multiple `<p>`, some `<ul>`, etc.

The table above already states the section is  optional, so no need to repeat it here.

File: doc/main.html [hunk 3]
```diff
@@ -148,50 +150,103 @@ <h4>Overall structure</h4>
...
+
+        <p>The <dfn>Table of contents</dfn> contains a listing of all sections within the document and links to those sections.</p>
+
+        <p>No user text is required or allowed.</p>
+
+        <p>The heading and entire section will be generated automatically.</p>
+
+      </section>
+
+      <section id="sec-foreward-section">
```
Comment (palemieux) [inline doc/main.html:225]:
```suggestion
      <section id="sec-foreword-section">
```

File: doc/main.html [hunk 4]
```diff
@@ -148,50 +150,103 @@ <h4>Overall structure</h4>
...
+              <td>Conditional (if informative references present)</td>
             </tr>
           </tbody>
         </table>
 
+        <p class="note">For further section infomation and requirements outside of this document, see <a href="#bib-iso-directives-part2"></a> and <a href="#bib-smpte-ag-16"></a></p>
+
+      </section>
+
+      <section id="sec-toc-section">
```
Comment (palemieux) [inline doc/main.html:214]: I am not a big fan of having this here this since it is auto-generated.

File: doc/main.html [hunk 5]
```diff
@@ -148,50 +150,103 @@ <h4>Overall structure</h4>
...
+              <th>Mandatory, Conditional, or Optional</th>
             </tr>
           </thead>
           <tbody>
             <tr>
-              <td>0..1</td>
-              <td><a>introduction section</a></td>
+              <td>Unnumbered</td>
+              <td><a>Table of contents</a></td>
+              <td>Mandatory (generated automatically)</td>
```
Comment (palemieux) [inline doc/main.html:162]:
```suggestion
              <td>Automatically generated</td>
```

File: doc/main.html [hunk 6]
```diff
@@ -148,50 +150,103 @@ <h4>Overall structure</h4>
...
+            </tr>
+            <tr>
+              <td>Unnumbered</td>
```
Comment (palemieux) [inline doc/main.html:165]: Why is the numbering important if it is generated automatically?

File: doc/main.html [hunk 7]
```diff
@@ -148,50 +150,103 @@ <h4>Overall structure</h4>
...
+        <p>No user text is required or allowed.</p>
+
+        <p>The heading and entire section will be generated automatically.</p>
+
+      </section>
+
+      <section id="sec-foreward-section">
+        <h4>Forward section</h4>
+
+        <p>The <dfn>Foreward</dfn> contains background info about SMPTE, copyright notices, and additional infomation based on the document type (<code>pubType</code>).</p>
```
Comment (palemieux) [inline doc/main.html:228]:
```suggestion
        <p>The <dfn>Foreword</dfn> SMPTE boilerplate information based on the document type (<code>pubType</code>).</p>
```

File: doc/main.html [hunk 8]
```diff
@@ -148,50 +150,103 @@ <h4>Overall structure</h4>
...
+
+        <p>User text for additional infomation relevant to the document is allowed in addition to the general boilerplate. If adding additional text, the section shall be present as:</p>
```
Comment (palemieux) [inline doc/main.html:230]:
```suggestion
        <p>It can also contain author-specified text, e.g. list of substantive changes since the last edition. Author-specified text is added as follows:</p>
```

File: doc/main.html [hunk 9]
```diff
@@ -148,50 +150,103 @@ <h4>Overall structure</h4>
...
+
+<pre>&lt;section id=&quot;sec-foreward&quot;&gt;
+  &lt;p&gt;This is the additional infomation relevant to the document.&lt;/p&gt;
+&lt;/section&gt;
+</pre>
+        <p>The <code>id</code> attribute shall be present and equal to <code>sec-foreward</code></p>
```
Comment (palemieux) [inline doc/main.html:236]:
```suggestion
        <p>The <code>id</code> attribute shall be present and equal to <code>sec-foreword</code></p>
```

File: doc/main.html [hunk 10]
```diff
@@ -148,50 +150,103 @@ <h4>Overall structure</h4>
...
+
+        <p>If no additional user text is required, the section shall not be present.</p>
```
Comment (palemieux) [inline doc/main.html:238]: I am not sure this is needed since we do not really care if the `<section>` element is empty.

File: doc/main.html [hunk 11]
```diff
@@ -148,50 +150,103 @@ <h4>Overall structure</h4>
...
+
+        <p>The heading and boilerplate of the section will be generated automatically.</p>
```
Comment (palemieux) [inline doc/main.html:240]:
```suggestion
        <p>The heading of the section is generated automatically.</p>
```

File: doc/main.html [hunk 12]
```diff
@@ -200,13 +255,17 @@ <h4>introduction section</h4>
...
       </section>
 
       <section id="sec-scope-section">
-        <h4>Scope</h4>
+        <h4>Scope section</h4>
+
+        <p>The <dfn>Scope</dfn> contains the scope of the document.</p>
 
-        <p>The <dfn>scope section</dfn> contains the scope of the document.</p>
+        <p>The section shall be present as: </p>
```
Comment (palemieux) [inline doc/main.html:268]: I do not think we can say "shall" since it can contain more than one `<p>`

File: doc/main.html [hunk 13]
```diff
@@ -218,10 +277,11 @@ <h4>Scope</h4>
       </section>
 
       <section id="sec-conformance-section">
-        <h4>Conformance</h4>
+        <h4>Conformance section</h4>
 
-        <p>The <dfn>conformance section</dfn> contains conformance requirements and definitions, beyond the usual SMPTE conformance
-        statements.</p>
+        <p>The <dfn>Conformance</dfn> contains conformance requirements and definitions.</p>
```
Comment (palemieux) [inline doc/main.html:282]: _The conformance_ does not read well.

Why did you change it from _the conformance section_?

File: doc/main.html [hunk 14]
```diff
@@ -218,10 +277,11 @@ <h4>Scope</h4>
...
+
+        <p>User text beyond the usual SMPTE conformance statements are allowed in addition to the general boilerplate. If adding additional text, the section shall be present as:</p>
```
Comment (palemieux) [inline doc/main.html:284]:
```suggestion
        <p>If the conformance section is present as follows, any text within will be added to usual SMPTE conformance boilerplate.</p>
```

File: doc/main.html [hunk 15]
```diff
@@ -230,50 +290,63 @@ <h4>Conformance</h4>
 
         <p>The <code>id</code> attribute shall be present and equal to <code>sec-conformance</code>.</p>
 
+        <p>If no additional user text is required, the section shall not be present.</p>
```
Comment (palemieux) [inline doc/main.html:293]: See above.

File: doc/main.html [hunk 16]
```diff
@@ -230,50 +290,63 @@ <h4>Conformance</h4>
...
-      &lt;cite id=&quot;bib-ietf-rfc-3986&quot;&gt;IETF RFC 3986&lt;/cite&gt;, Uniform Resource Identifiers (URI): Generic Syntax,
-      T. Berners-Lee, et al., January 2005. &lt;a&gt;http://www.ietf.org/rfc/rfc3986.txt&lt;/a&gt;
+      &lt;cite id=&quot;bib-HTML-5&quot;&gt;HTML Standard&lt;/cite&gt;, Living Standard. 
+        &lt;a&gt;https://html.spec.whatwg.org/multipage/&lt;/a&gt;&lt;/li&gt;
     &lt;/li&gt;
   &lt;/ul&gt;
 &lt;/section&gt;</pre>
 
-        <p>The <code>id</code> attribute shall be present on the <code>section</code> element and equal to
+        <p>The <code>id</code> attribute shall be present and equal to
```
Comment (palemieux) [inline doc/main.html:315]: Why not specify where is it present?

**Review by palemieux — CHANGES_REQUESTED** [review]

File: doc/main.html (hunk 1 above)
Comment (SteveLLamb) [reply doc/main.html:258]: Wasn't sure how the validation and build worked, noted.

File: doc/main.html (hunk 2 above)
Comment (SteveLLamb) [reply doc/main.html:249]: Agreed on the "optional" and example. Others sections I have update present to formatted where applicable.

File: doc/main.html (hunk 3 above)
Comment (SteveLLamb) [reply doc/main.html:225]: Noted, my bad.

File: doc/main.html (hunk 4 above)
Comment (SteveLLamb) [reply doc/main.html:214]: Agreed, removed.

File: doc/main.html (hunk 5 above)
Comment (SteveLLamb) [reply doc/main.html:162]: Agreed.

File: doc/main.html (hunk 6 above)
Comment (SteveLLamb) [reply doc/main.html:165]: This is for informative purposely only, really meant to show user in the order listed. I have added "(informative)" to the header.

File: doc/main.html (hunk 7 above)
Comment (SteveLLamb) [reply doc/main.html:228]: Great, was looking for better wording for this.

File: doc/main.html (hunk 8 above)
Comment (SteveLLamb) [reply doc/main.html:230]: Much better wording, thanks!

File: doc/main.html (hunk 9 above)
Comment (SteveLLamb) [reply doc/main.html:236]: Noted.

File: doc/main.html (hunk 10 above)
Comment (SteveLLamb) [reply doc/main.html:238]: I have changed this same as other comment above about absense of section.

File: doc/main.html (hunk 11 above)
Comment (SteveLLamb) [reply doc/main.html:240]: Changed as other sections for consistency.

File: doc/main.html (hunk 12 above)
Comment (SteveLLamb) [reply doc/main.html:268]: Changed as other sections for consistency.

File: doc/main.html (hunk 13 above)
Comment (SteveLLamb) [reply doc/main.html:282]: Oops. I removed section from the definitions (as shown in the table), didn't mean to remove from the descriptions. Updated.

File: doc/main.html (hunk 14 above)
Comment (SteveLLamb) [reply doc/main.html:284]: Much better, thanks again.

File: doc/main.html (hunk 15 above)
Comment (SteveLLamb) [reply doc/main.html:293]: Yes, updated.

File: doc/main.html (hunk 16 above)
Comment (SteveLLamb) [reply doc/main.html:315]: I think that was a bad copy/paste. Thanks.

**Comment by SteveLLamb** [comment]
@palemieux All changes requested updated in https://github.com/SMPTE/html-pub/pull/65/commits/0e8873d9a31cbefd0bdee3c6c591d2d447219dab
